[#2] fix: Input validation for component maximum score, weight, and name

jeffverz opened the pull request:
I have updated the feature [#2] to include input validation. Kindly check and approve merge to main branch.

Thanks.

jeffverz committed:
[#2] fix: Input validation for component maximum score, weight, and name

max172-hqt commented on src/components/EditGradeModal.tsx:
```diff
@@ -72,7 +84,13 @@ export default function EditGradeModal({
+                value={weight}
+                w={24}
+                onChangeText={handleWeightChange}
+                maxLength={2}
```
Can't enter float number.

max172-hqt commented on src/components/EditGradeModal.tsx:
```diff
@@ -29,9 +29,21 @@ export default function EditGradeModal({
+    }
+
+    if (!isNaN(+inputName)) {
+      Alert.alert('Invalid Input', 'Name should describe the component and not numeric only');
```
I think "Name should be string and cannot be empty" is more concise.

max172-hqt commented on src/components/EditGradeModal.tsx:
```diff
@@ -29,9 +29,21 @@ export default function EditGradeModal({
-      Alert.alert('Error', 'Input grade is not valid. Please try again.');
+      Alert.alert('Error', 'Input value must be numeric. Please try again.');
+      return;
+    } else if (Math.sign(inputWeight) === -1 || Math.sign(inputMaxScore) === -1) {
```
No need `else if` if you return early.

max172-hqt commented on src/components/EditGradeModal.tsx:
```diff
@@ -29,9 +29,21 @@ export default function EditGradeModal({
   const handleSavePressed = () => {
     const inputMaxScore = parseFloat(maxScore);
     const inputWeight = parseFloat(weight);
+    const inputName = name;
```
Why do we need a new variable?

max172-hqt reviewed (changes requested):
Can we also have upper-bound for these value, so that user can't enter sone ridiculous number. Example, name cannot be longer than 256 characters, weight < 100, score < 1000.

jeffverz committed:
[#2] fix: Updated input validation for grade component's name

Set to allow only numbers, letters and space.

jeffverz committed:
[#2] fix: Updated  some validation messages

-removed unnecessary  lines.
-returned conditional check to individual instead of nested if-else
-maxLength parameter for input Weight removed

max172-hqt committed:
[#2] fix: Weight upper bound
